docs/arcaflow/getting-started.md

```diff
@@ -105,142 +189,488 @@ input:
+
+```yaml title="example plugin schema YAML"
+steps:
+  hello-world:
+    display:
+      description: Says hello :)
+      name: Hello world!
+    id: hello-world
+    input:
+      objects:
+        FullName:
+          id: FullName
+          properties:
+            first_name:
+              display:
+                name: First name
+              examples:
+              - '"Arca"'
+              required: true
+              type:
+                min: 1
+                pattern: ^[a-zA-Z]+$
+                type_id: string
+            last_name:
+              display:
+                name: Last name
+              examples:
+              - '"Lot"'
+              required: true
+              type:
+                min: 1
+                pattern: ^[a-zA-Z]+$
+                type_id: string
+        InputParams:
+          id: InputParams
+          properties:
+            name:
+              display:
+                description: Who do we say hello to?
+                name: Name
+              examples:
+              - '{"_type": "fullname", "first_name": "Arca", "last_name": "Lot"}'
+              - '{"_type": "nickname", "nick": "Arcalot"}'
+              required: true
+              type:
+                discriminator_field_name: _type
+                type_id: one_of_string
+                types:
+                  fullname:
+                    display:
+                      name: Full name
+                    id: FullName
+                    type_id: ref
+                  nickname:
+                    display:
+                      name: Nick
+                    id: Nickname
+                    type_id: ref
+        Nickname:
+          id: Nickname
+          properties:
+            nick:
+              display:
+                name: Nickname
+              examples:
+              - '"Arcalot"'
+              required: true
+              type:
+                min: 1
+                pattern: ^[a-zA-Z]+$
+                type_id: string
+      root: InputParams
+    outputs:
+      error:
+        error: false
+        schema:
+          objects:
+            ErrorOutput:
+              id: ErrorOutput
+              properties:
+                error:
+                  display: {}
+                  required: true
+                  type:
+                    type_id: string
+          root: ErrorOutput
+      success:
+        error: false
+        schema:
+          objects:
+            SuccessOutput:
+              id: SuccessOutput
+              properties:
+                message:
+                  display: {}
+                  required: true
+                  type:
+                    type_id: string
+          root: SuccessOutput
+```
+
+The plugin schema can also be returned in JSON format, in which case you must specify wheter to return either the `input` or `output` schema.
 
```

Comment:
s/wheter/whether/

And, given that we're using "whether", I would omit "either".